File: main/frontend/lib/src/pages/shared/toggle_button/toggle_button_component.dart
import 'dart:async';

import 'package:ng_bootstrap/ng_bootstrap.dart';
import 'package:angular/angular.dart';

@Component(
  selector: 'toggle-button',
  templateUrl: 'toggle_button_component.html',
  styleUrls: const ['toggle_button_component.css'],
  directives: const [coreDirectives, BsTooltipComponent],
)
class ToggleButtonComponent {
  
  @Input("enabled")
  bool enabled;
  
  final changedSC = new StreamController();
  @Output() Stream get changed => changedSC.stream;
  
  ToggleButtonComponent() {
  	enabled = false;
  }
}

File: cinco-cloud/services/main/frontend/src/app/core/components/toggle-button/toggle-button.component.ts
import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'cc-toggle-button',
  templateUrl: './toggle-button.component.html',
  styleUrls: ['./toggle-button.component.scss']
})
export class ToggleButtonComponent {

  @Input()
  enabled = false;

  @Input()
  enabledTooltip = null;

  @Input()
  disabledTooltip = null;

  @Output()
  toggle = new EventEmitter<boolean>();
}
